components/script/dom/domexception.rs

```diff
@@ -105,11 +109,64 @@ impl DOMExceptionMethods for DOMException {
             DOMErrorName::NotReadableError => "The I/O read operation failed.",
         };
 
-        DOMString::from(message)
+        (
+            DOMString::from(message),
+            DOMString::from(format!("{:?}", code)),
+        )
+    }
+
+    fn new_inherited(message_: DOMString, name_: DOMString) -> DOMException {
+        DOMException {
+            reflector_: Reflector::new(),
+            message: message_,
+            name: name_,
+        }
+    }
+
+    pub fn new(global: &GlobalScope, code: DOMErrorName) -> DomRoot<DOMException> {
+        let (message, name) = DOMException::get_error_data_by_code(code);
+
+        reflect_dom_object(
+            Box::new(DOMException::new_inherited(message, name)),
+            global,
+            DOMExceptionBinding::Wrap,
+        )
+    }
+
+    pub fn Constructor(
+        global: &GlobalScope,
+        message: DOMString,
+        name: DOMString,
+    ) -> Result<DomRoot<DOMException>, Error> {
+        Ok(reflect_dom_object(
+            Box::new(DOMException::new_inherited(message, name)),
+            global,
+            DOMExceptionBinding::Wrap,
+        ))
+    }
+}
+
+impl DOMExceptionMethods for DOMException {
+    // https://heycam.github.io/webidl/#dfn-DOMException
+    fn Code(&self) -> u16 {
+        match DOMErrorName::from(&self.name) {
+            Some(code) => code as u16,
+            None => 0 as u16,
+        }
+    }
+
+    // https://heycam.github.io/webidl/#idl-DOMException-error-names
+    fn Name(&self) -> DOMString {
+        self.name.clone()
+    }
+
+    // https://heycam.github.io/webidl/#error-names
```

Comment:
Should this be https://heycam.github.io/webidl/#dom-domexception-message instead?